lib/rust/ensogl/component/text/src/font/glyph.rs

```diff
@@ -44,79 +35,163 @@ pub type Texture = gpu::Texture<texture::GpuOnly, texture::Rgb, u8>;
+
+    /// Color getter.
     pub fn color(&self) -> Rgba {
         self.color.get().into()
     }
 
+    /// Color setter.
     pub fn set_color(&self, color: impl Into<Rgba>) {
         self.color.set(color.into().into())
     }
 
-    pub fn is_bold(&self) -> bool {
-        self.style.get() & style_flag::BOLD != 0
-    }
-
-    pub fn set_bold(&self, value: bool) {
-        self.style.modify(|v| if value { *v |= style_flag::BOLD } else { *v &= !style_flag::BOLD });
-    }
-
+    /// SDF-based glyph thickness adjustment. Values greater than 0 make the glyph thicker, while
+    /// values lower than 0 makes it thinner.
     pub fn sdf_bold(&self) -> f32 {
         self.sdf_bold.get()
     }
 
+    /// SDF-based glyph thickness getter.
     pub fn set_sdf_bold(&self, value: f32) {
         self.sdf_bold.set(value);
     }
 
+    /// Size getter.
     pub fn font_size(&self) -> f32 {
         self.font_size.get()
     }
 
+    /// Size setter.
     pub fn set_font_size(&self, size: f32) {
         self.font_size.set(size);
-        let glyph_info = self.font.glyph_info(self.char.get());
-        self.sprite.size.set(glyph_info.scale.scale(size));
+        let opt_glyph_info = self.font.glyph_info(
+            self.properties.get(),
+            &self.variations.borrow(),
+            self.glyph_id.get(),
+        );
+        if let Some(glyph_info) = opt_glyph_info {
+            self.sprite.size.set(glyph_info.scale.scale(size))
+        }
     }
 
     /// Change the displayed character.
     pub fn set_char(&self, ch: char) {
-        self.char.set(ch);
-        let glyph_info = self.font.glyph_info(ch);
-        self.atlas_index.set(glyph_info.msdf_texture_glyph_id as f32);
-        self.update_msdf_texture();
-        let font_size = self.font_size();
-        self.sprite.size.set(glyph_info.scale.scale(font_size));
-    }
-
-    // FIXME: How does it work? Replace with better checking.
-    fn update_msdf_texture(&self) {
-        let texture_changed = self.atlas.with_content(|texture| {
-            texture.storage().height != self.font.msdf_texture_rows() as i32
-        });
+        let opt_glyph_id =
+            self.font.glyph_id_of_code_point(self.properties.get(), &self.variations.borrow(), ch);
+        if let Some(glyph_id) = opt_glyph_id {
+            self.set_glyph_id(glyph_id)
+        }
+        // FIXME: display not found char
+    }
+
+    /// Change the displayed character.
+    pub fn set_glyph_id(&self, glyph_id: GlyphId) {
+        self.glyph_id.set(glyph_id);
+        let opt_glyph_info =
+            self.font.glyph_info(self.properties.get(), &self.variations.borrow(), glyph_id);
+        if let Some(glyph_info) = opt_glyph_info {
+            self.atlas_index.set(glyph_info.msdf_texture_glyph_id as f32);
+            self.update_atlas();
+            let font_size = self.font_size();
+            self.sprite.size.set(glyph_info.scale.scale(font_size));
+        }
```

Comment:
If there is no glyph with given id, we silently do nothing (something we should not to do), still keeping the new id in `glyph_id` field.